src/Compilers/CSharp/Portable/Symbols/Source/SourceConstructorSymbol.cs

```diff
@@ -42,7 +46,7 @@ internal sealed class SourceConstructorSymbol : SourceMethodSymbol
                     diagnostics.Add(ErrorCode.ERR_ExternHasConstructorInitializer, location, this);
                 }
 
-                if (syntax.Body != null)
+                if (hasBlockBody || IsExpressionBodied)
                 {
                     diagnostics.Add(ErrorCode.ERR_ExternHasBody, location, this);
```

Comment:
Yes, you've chosen to modify the condition, there should be a test that proves that that was necessary and to ensure the change won't be undone by accident.